### ISSUE TITLE
 Pre-training AMPLIFY 350M on a Single Machine with 2 GPUs= >main.py Not found issue

### ISSUE THREAD

**Issue by Ariya-f** [chandar-lab/AMPLIFY#11]
Hi,
I have been trying to use AMPLIFY recently.
I successfully installed it and confirmed the installation.

However, when I try to run the pre-training of AMPLIFY 350M on a single machine, I encounter the following error:
/py310/bin/python3.10: can't open file '/home/AMPLIFY/main.py': [Errno 2] No such file or directory

when I serarch the cloned AMPLIFY reposetory, I cant find the main.py

I am completely new to this field and an enthusiast, so please excuse my amateur questions!
I can’t find any main.py. Is it supposed to be trainer.py instead?

**Comment by qfournier** [comment]
Hi, 

Thank you for pointing out the issue! I have updated the documentation. In addition, here is the exact self-contained example that I ran on my end.

Clone and install AMPLIFY from source:
```bash
git clone git@github.com:chandar-lab/AMPLIFY.git
cd AMPLIFY
python3 -m venv env
source env/bin/activate
python3 -m pip install --upgrade pip
python3 -m pip install --editable .[dev]
python3 -m pytest
```

Download, extract, and pre-process the datasets:
``` bash
wget https://zenodo.org/records/13834052/files/uniref100_train.fasta.zip -P $SCRATCH
unzip $SCRATCH/uniref100_train.fasta.zip -d $SCRATCH
python scripts/fasta_to_csv.py -i $SCRATCH/uniref100_train.fasta -o $SCRATCH/uniref100_train.csv

wget https://zenodo.org/records/13834052/files/uniprot_dev.fasta.zip -P $SCRATCH
unzip $SCRATCH/uniprot_dev.fasta.zip -d $SCRATCH
python scripts/fasta_to_csv.py -i $SCRATCH/uniprot_dev.fasta -o $SCRATCH/uniprot_dev.csv

wget https://zenodo.org/records/13834052/files/scop_dev.fasta.zip -P $SCRATCH
unzip $SCRATCH/scop_dev.fasta.zip -d $SCRATCH
python scripts/fasta_to_csv.py -i $SCRATCH/scop_dev.fasta -o $SCRATCH/scop_dev.csv

wget https://zenodo.org/records/13834052/files/oas_dev.fasta.zip -P $SCRATCH
unzip $SCRATCH/oas_dev.fasta.zip -d $SCRATCH
python scripts/fasta_to_csv.py -i $SCRATCH/oas_dev.fasta -o $SCRATCH/oas_dev.csv
```

Launch the job:
```bash
accelerate launch \
	--config_file=conf/accelerate_ddp.yaml \
	--num_processes=1 \
	--mixed_precision=fp16 \
	--gradient_clipping=1.0 \
	scripts/pretrain.py \
	hydra.run.dir=logs/AMPLIFY_350M \
	wandb.dir=logs/AMPLIFY_350M \
	wandb.name=AMPLIFY_350M \
	dataset=uniref100 \
	dataset.train.paths.uniref100=$SCRATCH/uniref100_train.csv \
	dataset.validation.paths.uniprot=$SCRATCH/uniprot_dev.csv \
	dataset.validation.paths.pdb=$SCRATCH/scop_dev.csv \
	dataset.validation.paths.oas=$SCRATCH/oas_dev.csv \
	trainer.dir=logs/AMPLIFY_350M \
	trainer.train.per_device_batch_size=32 \
	trainer.validation.per_device_batch_size=32 \
	trainer.gradient_accumulation_steps=1
```
Note that:
- `num_processes=1` because my machine has a single GPU
- `mixed_precision=fp16` because the GPU is old and does not support `bf16`
- `gradient_accumulation_steps=1` because it is faster for debugging

**Comment by Ariya-f** [comment]
Hi,
Thank you so much for providing the additional info and corrections.
I was able to run AMPLIFY on one GPU using your "accelerate launch" setup.
gradient_accumulation_steps=1 was very helfull too.
I wonder, though, how long it would usually take to train it on UniRef100.

**Comment by qfournier** [comment]
Great, I'm glad it worked on your end! 

AMPLIFY was trained with a batch size of 2 million tokens, or 4,096 sequences of length 512 in Stage 1. Depending on the number of GPUs and their memory capacity, gradient accumulation may be necessary to achieve this batch size. Note that batch size plays a significant role in the model's performance.

AMPLIFY 120M and 350M were trained for one month using 1 x A100 DGX (8 x A100 80Gb) and 4 x A100 DGXs (32 x A100 80Gb), respectively. The pre-training cost is why we are releasing the checkpoints! 

Note that you can fine-tune the models much more easily. A single A100 for an hour is enough to fine-tune a 2-layer MLP on top of AMPLIFY 350M for downstream tasks.
